include/ccf/http_etag.h

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+#pragma once
+
+#define FMT_HEADER_ONLY
+#include <regex>
+#include <set>
+#include <string>
+
+namespace ccf
+{
+  namespace http
+  {
+    /** Utility class to resolve If-Match and If-None-Match as described
+     * in https://www.rfc-editor.org/rfc/rfc9110#field.if-match
+     */
+    class Matcher
+    {
+    private:
+      /// If-Match header is not present
+      bool _empty = false;
+      /// If-Match header is present and has the value "*"
+      bool any_value = false;
+      /// If-Match header is present and has specific etag values
+      std::set<std::string> if_etags;
+
+    public:
+      /** Construct a Matcher from a match header if present
+       *
+       * Note: Weak tags are not supported.
+       */
+      Matcher(const std::optional<std::string>& match_header)
```

Comment:
Could this take a `string` rather than an `optional`, _only_ be constructed when the header was present, and then this matcher could drop all of its `empty` state/code? I think all of the paths are supported identical (checks on `.empty()` on a `optional<string>` rather than a `Matcher`, an error if both _headers_ are present, errors thrown if the content is not a valid ETags list), just in maybe different orders.

Reply:
Probably, I started that way but this seemed to reduce the number of branches somewhat. I'll have another look.